rules/xcodeproj.bzl

```diff
@@ -268,10 +308,24 @@ def _xcodeproj_impl(ctx):
                 fi = "$BAZEL_WORKSPACE_ROOT/%s" % fi
             framework_search_paths.append("\"%s\"" % fi)
         target_settings["FRAMEWORK_SEARCH_PATHS"] = " ".join(framework_search_paths)
-        target_settings["GCC_PREPROCESSOR_DEFINITIONS"] = " ".join(["\"%s\"" % d for d in target_info.cc_defines.to_list()])
+        macros = ["\"%s\"" % d for d in target_info.cc_defines.to_list()]
+        macros.append("$(inherited)")
+        target_settings["GCC_PREPROCESSOR_DEFINITIONS"] = " ".join(macros)
+
         if target_info.product_type == "application":
             target_settings["INFOPLIST_FILE"] = "$BAZEL_STUBS_DIR/Info-stub.plist"
             target_settings["PRODUCT_BUNDLE_IDENTIFIER"] = target_info.bundle_id
+        else:
+            defines_without_equal_sign = ["$(inherited)"]
+            for d in target_info.swift_defines.to_list():
+                defines_without_equal_sign.append(d)
```

Comment:
I think line 321 should not be here. Or else you will end up appending everything.